### PR TITLE
feat: show alerts when global conf generation is not working properly

### PR DESCRIPTION
**Pull request by justasnortal**
Refs: XRDDEV-2338

### CODE REVIEW

File: src/central-server/admin-service/core/src/main/java/org/niis/xroad/cs/admin/core/service/NotificationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    private final GlobalConfGenerationStatusService globalConfGenerationStatus;
+
+    @Override
+    public Set<AlertInfo> getAlerts() {
+
+        final List<TokenInfo> tokens;
+        try {
+            tokens = signerProxyFacade.getTokens();
+        } catch (Exception e) {
+            return Set.of(new AlertInfo("status.signer_error"));
```
Comment (ricardas-buc) [inline src/central-server/admin-service/core/src/main/java/org/niis/xroad/cs/admin/core/service/NotificationServiceImpl.java:75]: Is it OK to hide root exception? Maybe log it?

File: src/central-server/admin-service/core/src/main/java/org/niis/xroad/cs/admin/core/service/NotificationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        return Set.of();
+    }
+
+    private Set<AlertInfo> checkConfigurationSigningKey(String sourceType, List<TokenInfo> tokens) {
+        final Optional<ConfigurationSigningKey> signingKeyOptional = configurationSigningKeysService.findActiveForSource(sourceType);
+        if (signingKeyOptional.isEmpty()) {
+            return Set.of(new AlertInfo(format("status.signing_key.%s.missing", sourceType)));
+        }
+        final ConfigurationSigningKey signingKey = signingKeyOptional.get();
+        for (final TokenInfo token : tokens) {
```
Comment (ricardas-buc) [inline src/central-server/admin-service/core/src/main/java/org/niis/xroad/cs/admin/core/service/NotificationServiceImpl.java:124]: did this pass sonarqube complexity check? Might be hard to simplify, but worth considering.

File: src/central-server/admin-service/core/src/main/java/org/niis/xroad/cs/admin/core/service/NotificationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+    private boolean softwareTokenInitialized(List<TokenInfo> tokens) {
+        return tokens.stream()
+                .anyMatch(token -> token.getId().equals(SOFTWARE_TOKEN_ID) && token.getStatus() != NOT_INITIALIZED);
+    }
+
+    private Set<AlertInfo> checkGlobalConfGenerationStatus() {
+        final Optional<GlobalConfGenerationStatus> confStatus = globalConfGenerationStatus.get();
+
+        if (confStatus.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (justasnortal) [reply src/central-server/admin-service/core/src/main/java/org/niis/xroad/cs/admin/core/service/NotificationServiceImpl.java:106]: added logging.

**Comment by justasnortal** [comment]
[rerun-ci]

File: src/central-server/admin-service/core/src/main/java/org/niis/xroad/cs/admin/core/service/NotificationServiceImpl.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (justasnortal) [reply src/central-server/admin-service/core/src/main/java/org/niis/xroad/cs/admin/core/service/NotificationServiceImpl.java]: Added unknown status.

**Review by ricardas-buc — APPROVED** [review]

**Pull request merged**